client/app/scripts/superdesk-search/search.js

```diff
@@ -1020,6 +1026,62 @@
                             }, 0, false);
                         });
                     });
+
+                    metadata.fetchSubjectcodes().then(function () {
+                        scope.subjectcodes = metadata.values.subjectcodes;
+                        tags.initSelectedFacets().then(function (currentTags) {
+                            scope.subjectitems = {
+                                subject: getSubjectCodes(currentTags)
+                            };
+                        });
+                    });
+
+                    function getSubjectCodes(currentTags) {
+                        var queryArray = currentTags.selectedParameters, filteredArray = [];
+                        if (!$location.search().q) {
+                            return filteredArray;
+                        }
+                        for (var i = 0, queryArrayLength = queryArray.length; i < queryArrayLength; i++) {
+                            var queryArrayElement = queryArray[i];
+                            if (queryArrayElement.indexOf('subject.name') !== -1) {
+                                var elementName = queryArrayElement.substring(
+                                        queryArrayElement.lastIndexOf('(') + 1,
+                                        queryArrayElement.lastIndexOf(')')
+                                        );
+                                for (var j = 0, subjectCodesLenght = scope.subjectcodes.length; j < subjectCodesLenght; j++) {
```

Comment:
"lenght" is spelled "l-e-n-g-t-h" :wink: (seen it at a couple of places).

(don't need to change the PR just because of this, though, only consider changing it, if you need to apply some other changes as well)